src/Components/Components/src/ComponentsMetrics.cs

```diff
@@ -0,0 +1,239 @@
+
+    public bool IsBatchDurationEnabled => _batchDuration.Enabled;
+    public bool IsBatchExceptionEnabled => _batchException.Enabled;
+
+    public ComponentsMetrics(IMeterFactory meterFactory)
+    {
+        Debug.Assert(meterFactory != null);
+
+        _meter = meterFactory.Create(MeterName);
+        _lifeCycleMeter = meterFactory.Create(LifecycleMeterName);
+
+        _navigationCount = _meter.CreateCounter<long>(
+            "aspnetcore.components.navigation",
+            unit: "{route}",
+            description: "Total number of route changes.");
+
+        _eventDuration = _meter.CreateHistogram(
+            "aspnetcore.components.event.duration",
+            unit: "s",
+            description: "Duration of processing browser event.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _eventException = _meter.CreateCounter<long>(
+            "aspnetcore.components.event.exceptions",
+            unit: "{exception}",
+            description: "Total number of exceptions during browser event processing.");
+
+        _parametersDuration = _lifeCycleMeter.CreateHistogram(
+            "aspnetcore.components.update_parameters.duration",
+            unit: "s",
+            description: "Duration of processing component parameters.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _parametersException = _lifeCycleMeter.CreateCounter<long>(
+            "aspnetcore.components.update_parameters.exceptions",
+            unit: "{exception}",
+            description: "Total number of exceptions during processing component parameters.");
+
+        _batchDuration = _lifeCycleMeter.CreateHistogram(
+            "aspnetcore.components.rendering.batch.duration",
+            unit: "s",
+            description: "Duration of rendering batch.",
+            advice: new InstrumentAdvice<double> { HistogramBucketBoundaries = MetricsConstants.ShortSecondsBucketBoundaries });
+
+        _batchException = _lifeCycleMeter.CreateCounter<long>(
+            "aspnetcore.components.rendering.batch.exceptions",
+            unit: "{exception}",
+            description: "Total number of exceptions during batch rendering.");
+    }
+
+    public void Navigation(string componentType, string route)
+    {
+        var tags = new TagList
+        {
+            { "component.type", componentType ?? "unknown" },
+            { "route", route ?? "unknown" },
+        };
+
+        _navigationCount.Add(1, tags);
+    }
+
+    public async Task CaptureEventDurationAsync(Task task, long startTimestamp, string? componentType, string? methodName, string? attributeName)
+    {
+        try
+        {
+            await task;
+
+            var tags = new TagList
+            {
+                { "component.type", componentType ?? "unknown" },
```

Comment:
`code.class.name`? to match https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/code.md

Reply:
We don't really have code reference, as in file name and line number.

Reply:
My initial thought is that the thing being refered to was the Blazor component abstraction in which case component.type and component.method made sense to me. The fact that code.class.name isn't currently defined also nudges me in that direction. If we did want to use code.* attributes instead this feels like another spot where OTel semconv feedback could be helpful.